pandas/tests/test_frame.py

```diff
@@ -10755,7 +10755,11 @@ def test_filter(self):
         df = DataFrame(0., index=[0, 1, 2], columns=[0, 1, '_A', '_B'])
         filtered = df.filter(like='_')
         self.assertEqual(len(filtered.columns), 2)
-
+        
+        # regex with ints in column names
+        df = DataFrame(0., index=[0, 1, 2], columns=[0, 1, 'A1', 'B'])
+        filtered = df.filter(regex='^[0-9]+$')
+        self.assertEqual(len(filtered.columns), 2)
```

Comment:
do the test again with all number columns that are strings, e.g. ``['0','1'...]``  (i think results should be the same)

on the comparision do

```
expected = DataFrame(.....)
assert_frame_equal(fitlered, expected)
```

IOW construct the expected manually